clang/lib/Analysis/ThreadSafety.cpp

```diff
@@ -2282,6 +2308,9 @@ void ThreadSafetyAnalyzer::runAnalysis(AnalysisDeclContext &AC) {
   const NamedDecl *D = walker.getDecl();
   CurrentFunction = dyn_cast<FunctionDecl>(D);
 
+  if (isa<FunctionDecl, ObjCMethodDecl>(D))
```

Comment:
This shouldn't be needed. It should be always true, and even if it wasn't, the check doesn't rely on it.

Reply:
The `dyn_cast` right above is confusing then :thinking:

Reply:
The way I understand this, `isa<FunctionDecl, ObjCMethodDecl>` checks if the argument is a `FunctionDecl` or `ObjCMethodDecl`, and that is always true. The `dyn_cast` is to `FunctionDecl` specifically.